packages/braid-design-system/src/lib/components/TextLink/TextLink.tsx

```diff
@@ -102,9 +103,38 @@ export const useLinkStyles = ({
   );
 };
 
+export const TextLinkContent = ({
+  icon,
+  iconPosition = 'leading',
+  children,
+}: Pick<TextLinkProps, 'icon' | 'iconPosition' | 'children'>) => (
+  <>
+    {icon && iconPosition === 'leading' ? (
+      <Box component="span" paddingRight="xxsmall">
+        {icon}
+      </Box>
+    ) : null}
+    {children}
+    {icon && iconPosition === 'trailing' ? (
+      <Box component="span" paddingLeft="xxsmall">
+        {icon}
+      </Box>
+    ) : null}
```

Comment:
Do we want make sure the icon doesn't break on a new line?

<img width="339" alt="wrap" src="https://user-images.githubusercontent.com/3297808/223601184-de4820cb-993b-49d5-a161-a10ff9c9c9a8.png">

Reply:
Upon further testing, the above suggestion doesn't actually work 🤔

Reply:
Yeah its missing the `nowrap` rule on the span. All good i'm working standardising that solution across the multiple usages now.

Reply:
I was playing around and must have added the `nowrap` at some point to make it work. Happy for you to take it from here 👍